CONTRIBUTING.md

```diff
@@ -3,31 +3,126 @@
 ## Overview
 
 The AdoptOpenJDK API V3 is a Kotlin application (fronted by Swagger and OpenAPI) which makes 
-calls via the GitHub API inb order to retrieve AdoptOpenJDK binaries and metadata.
+calls via the GitHub API in order to retrieve AdoptOpenJDK binaries and metadata.
 
 Since the GitHub API is rate limited we use MongoDB as a caching mechanism.
 
+## Source code management and branching
+
+We treat the AdoptOpenJDK org repository as the canonical repository for deploying the API from. 
+We use the `staging` branch to trial any changes in a Production like environment and then 
+eventually merge into `master` for a real Production deployment.
+
+**NOTE** Please ensure for any significant change that you Pull Request to the `staging` branch 
+
 ## Build
 
 ### Pre-Requisites
 
-Java 11 is required to build.
+Java 11 is a requirement to build to project.
 
 ### Build Tool
 
-We use a Maven Wrapper (mvnw) to ensure that there's a consistent repeatable build.
+We use a Maven Wrapper (mvnw) to ensure that there's a consistent, repeatable build. The 
+[POM File](./pom.xml) is the place to start.
+
+**NOTE:** We use a multi-module project structure. The root level POM contains a majority 
+of the configuration that the children inherit from. 
+
+### Build Command
 
-### Build
+To perform a full build and test you run the following:
 
 `./mvnw clean install`
 
-## Usage
+If you wish to view all fo the Maven reporting about the project you run the following:
+
+`./mvnw clean install site`
+
+## Testing
+
+**WARN** This API is critical to the success of AdoptOpenJDK, so we have a strong preference that 
```

Comment:
Rather than suggesting that it's a strong preference we could be more explicit by saying that something like

> **WARN** This API is critical to the success of AdoptOpenJDK therefore it is essential that tests are provided for all new functionality.